readthedocs/gold/views.py

```diff
@@ -1,146 +1,90 @@
-import datetime
+"""Gold suscription views"""
 
-from django.core.urlresolvers import reverse
+import stripe
+from django.core.urlresolvers import reverse, reverse_lazy
 from django.conf import settings
-from django.db import IntegrityError
-from django.http import HttpResponseRedirect, Http404
+from django.contrib.messages.views import SuccessMessageMixin
+from django.contrib import messages
+from django.http import HttpResponseRedirect
 from django.shortcuts import render_to_response, get_object_or_404
 from django.template import RequestContext
 from django.contrib.auth.decorators import login_required
+from django.utils.translation import ugettext_lazy as _
+from vanilla import DeleteView, UpdateView, DetailView
 
-import stripe
+from readthedocs.core.mixins import LoginRequiredMixin
+from readthedocs.projects.models import Project
+from readthedocs.payments.mixins import StripeMixin
 
-from .forms import CardForm, GoldProjectForm
+from .forms import GoldSubscriptionForm, GoldProjectForm
 from .models import GoldUser
-from readthedocs.projects.models import Project
 
 stripe.api_key = settings.STRIPE_SECRET
```

Comment:
Yeah, i just copied what we had. This could move into `StripeResourceMixin` no problem, or set before we use a call to Stripe on one-off functions.